### ISSUE TITLE
Warn against statements like (a = 3)if a == 2

### ISSUE THREAD

**Issue by daviddavis** [rubocop/rubocop#467]
I was surprised to see that rubocop did not throw a convention error given the following line of code. It looks like there is no cop to handle this. Note the lack of whitespace between `)` and `if`.

``` ruby
(a = 3)if a == 2
```

I'd be happy to work on the cop for this but I don't know if the rule should be:
1. Require whitespace after closing parenthesis
2. Require whitespace before a modifier

Thoughts?

**Edit:** Found some more (similar) statements that pass rubocop:

``` ruby
a = "test"if a == 2
a = 42unless a == 2
a = [1,2,3]if a == 2
a = {:a => "b"}if a == 2
```
